flybywiresim/aircraft
# fix(CDU): suppress TMPY FPLN when no modification made in airways page

Fixes #8910 

## Summary of Changes

Corrected FMS logic to prevent TMPY FPLN from appearing after leaving airways page without making any modifications.

## Screenshots (if necessary)


Refer to #8910 for previous incorrect behaviour

Corrected:
https://github.com/user-attachments/assets/76c8f6e1-591a-43c0-9f33-521591564ba4

## References





## Additional context



Discord username (if different from GitHub):
robeet

## Testing instructions

1. Load anywhere, and start a flight plan from the MCDU
2. Open lateral revision page for a waypoint -> open airways page
3. Without making any modifications, press LSK6L (RETURN)
4. Should return to lat rev page -> press LSK6L (RETURN) again
5. Should return to flight plan page -> verify no TMPY FPLN state

Secondary:
1. Following step 3 above, return to airways page (LSK5R) and make a modification, then insert
2. Observe TMPY FPLN and confirm insert
3. Verify new modifications inserted correctly


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
